src/sinks/gcp/stackdriver_metrics.rs

```diff
@@ -1,3 +1,7 @@
+// TODO: In order to correctly assert component specification compliance, we would have to do some more advanced mocking
+// off the endpoint, which would include also providing a mock OAuth2 endpoint to allow for generating a token from the
+// mocked credentials. Let this TODO serve as a placeholder for doing that in the future.
```

Comment:
Noting here to make sure to add an issue for this TODO once this PR is merged.

Reply:
Opened #12856 to track this.